src/layers/masked_patch_prediction_layer.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import tensorflow as tf
+from official.modeling import tf_utils
+
+import tensor_utils
+
+
+class MaskedPP(tf.keras.layers.Layer):
+  """Masked patch prediction network head for image modeling.
+
+  This layer implements a masked patch prediction based on the provided
+  transformer based encoder.
+
+  Args:
+    output_num_classes: The number of output classes for patch prediction.
+      it should be (2 ** output_channel_bits ) ** 3.
+    activation: The activation, if any, for the dense layer.
+    initializer: The initializer for the dense layer. Defaults to a Glorot
+      uniform initializer.
+    output: The output style for this layer. Can be either 'logits' or
+      'predictions'.
+      
+  Inputs:
+    sequence_data: <tf.Tensor>[batch, sequence_length, embedding_size]
+      encoder sequence output.
+    masked_positions: <tf.Tensor>[batch, mask_sequence_length]
+      the indices of masked tokens.
+    
+  Outputs: <tf.Tensor>[batch, mask_sequence_length, output_num_classes] 
+    the logits or predictions.
+  """
+
+
```

Comment:
tiny nit: I think the Google style mandates just one blank line here instead of two.